contract/services.py

```diff
@@ -417,10 +555,40 @@ def contract_valuation(self, contract_contribution_plan_details):
                 ccpd_record = model_to_dict(ccpd)
                 ccpd_record["calculated_amount"] = calculated_amount
                 if contract_contribution_plan_details["save"]:
-                    ccpd.save(self.user)
-                    uuid_string = f"{ccpd.id}"
-                    ccpd_record['id'], ccpd_record['uuid'] = (uuid_string, uuid_string)
-                ccpd_list.append(ccpd_record)
+                    from core import datetime, datetimedelta
+                    length = get_contribution_length(cpbd.contribution_plan)
+                    ccpd.date_valid_from = contract_details["contract_date_valid_from"]
```

Comment:
add TODO: calculate the number of CCPD to create in order to cover the contract lenght